compute/apiv1/smoke_test.go

```diff
@@ -0,0 +1,257 @@
+// limitations under the License.
+
+// To run these tests, set GCLOUD_TESTS_GOLANG_PROJECT_ID env var to your GCP projectID
+
+package compute
+
+import (
+	"cloud.google.com/go/internal/testutil"
+	"cloud.google.com/go/internal/uid"
+	"context"
+	"fmt"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+	"testing"
+)
+
+var projectId = testutil.ProjID()
+var defaultZone = "us-central1-a"
+
+func TestCreateGetListInstance(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping smoke test in short mode")
+	}
+	space := uid.NewSpace("gogapic", nil)
+	name := space.New()
+	ctx := context.Background()
+	c, err := NewInstancesRESTClient(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	zonesClient, err := NewZoneOperationsRESTClient(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	diskType := computepb.AttachedDisk_PERSISTENT
+	instance := &computepb.Instance{
+		Name:        &name,
+		Description: proto.String("тест"),
+		MachineType: proto.String(fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/zones/%s/machineTypes/n1-standard-1", projectId, defaultZone)),
+		Disks: []*computepb.AttachedDisk{
+			{
+				AutoDelete: proto.Bool(true),
+				Boot:       proto.Bool(true),
+				Type:       &diskType,
+				InitializeParams: &computepb.AttachedDiskInitializeParams{
+					SourceImage: proto.String("projects/debian-cloud/global/images/family/debian-10"),
+				},
+			},
+		},
+		NetworkInterfaces: []*computepb.NetworkInterface{
+			{
+				AccessConfigs: []*computepb.AccessConfig{
+					{
+						Name: proto.String("default"),
+					},
+				},
+			},
+		},
+	}
+	createRequest := &computepb.InsertInstanceRequest{
+		Project:          projectId,
+		Zone:             defaultZone,
+		InstanceResource: instance,
+	}
+
+	insert, err := c.Insert(ctx, createRequest)
+	if err != nil {
+		t.Error(err)
+	}
+
+	waitZonalRequest := &computepb.WaitZoneOperationRequest{
+		Project:   projectId,
+		Zone:      defaultZone,
+		Operation: insert.GetName(),
+	}
+	_, err = zonesClient.Wait(ctx, waitZonalRequest)
+	if err != nil {
+		t.Error(err)
+	}
+	fmt.Printf("Inserted instance named %s\n", name)
+	defer ForceDeleteInstance(ctx, name, c)
+
+	getRequest := &computepb.GetInstanceRequest{
+		Project:  projectId,
+		Zone:     defaultZone,
+		Instance: name,
+	}
+	get, err := c.Get(ctx, getRequest)
+	if err != nil {
+		t.Error(err)
+	}
+	testutil.Equal(name, get.GetName())
+	testutil.Equal("тест", get.GetDescription())
+	listRequest := &computepb.ListInstancesRequest{
+		Project: projectId,
+		Zone:    defaultZone,
+	}
+
+	list, err := c.List(ctx, listRequest)
+	if err != nil {
+		t.Error(err)
+	}
+	items := list.GetItems()
+	found := false
+	for _, element := range items {
+		if element.GetName() == name {
+			found = true
+		}
+	}
+	if !found {
+		t.Error("Couldn't find the instance in list response")
+	}
+
+	deleteInstanceRequest := &computepb.DeleteInstanceRequest{
```

Comment:
Isn't this done by the defer already?

Reply:
logic is:
1) we want to test delete() method here and the test fails if there are errors from API.
2) defer delete is for cases where we inserted an instance but test execution was aborted before it ended.